resolwe/flow/engine.py

```diff
@@ -0,0 +1,16 @@
+"""Abstract Workflow Engine."""
+
+
+class BaseEngine(object):
+    """Base class for all engines in Resolwe workflow."""
+
+    name = None
+
+    def __init__(self, manager, settings=None):
+        """Construct a Resolwe engine."""
+        self.manager = manager
+        self.settings = settings or {}
```

Comment:
But then all invocations would share the same instance? I know that we are not modifying `settings` but I think that using `None` and constructing a new dictionary on each invocation is good practice.